src/clientiface.cpp

```diff
@@ -247,10 +264,16 @@ void RemoteClient::GetNextBlocks (
 				Don't generate or send if not in sight
 				FIXME This only works if the client uses a small enough
 				FOV setting. The default of 72 degrees is fine.
+				Also retrieve a smaller view cone in the direction of the player's
+				movement.
+				(0.1 is about 4 degrees)
 			*/
-
 			f32 dist;
-			if (!isBlockInSight(p, camera_pos, camera_dir, camera_fov, d_blocks_in_sight, &dist)) {
+			if (!(isBlockInSight(p, camera_pos, camera_dir, camera_fov,
+						d_blocks_in_sight, &dist) ||
+				(playerspeed.getLength() > 1.0f * BS &&
+				isBlockInSight(p, camera_pos, playerspeeddir, 0.1f,
```

Comment:
Lines 274 and 275 need 1 more indent to distinguish the condition logic from the action. This is something i missed in your previous commit.
Otherwise fine.

Reply:
Done. I find this far less readable than the longer line. But as usually I do not feel strongly. Let's get it pushed.